File: TowerDefence.spritebuilder/Map/DetachmentFabric.swift
//
//  DetachmentFabric.swift
//  TowerDefence
//
//  Created by koveres on 06.12.14.
//  Copyright (c) 2014 Apportable. All rights reserved.
//

import Foundation


public struct  DetachmentFabric {
    
    public static var detachments:[Detachment] = [Detachment]();
    public static  func  create( type__: SolderType, size: SizeGroupe ) -> (Detachment)
    {
        var newd:Detachment =  Detachment();
        newd.size = size;
        newd.type = type__;
        return newd;
    }
    public static func getDetachment( size_:SizeGroupe,  type_: SolderType) -> (Detachment)
    {
        for det in detachments
        {
    
        if(det.type == type_)
        {
            if(det.size ==  size_)
            {
            return det;
            }
        }
        }
        let newd:Detachment =  DetachmentFabric.create( type_,size:  size_);
        detachments.append(newd);
        return newd;
    }
}

File: test.spritebuilder/Source/TDEngine/Map/DetachmentFabric.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package Map;

import internalObject.SolderInternal;
import java.util.ArrayList;
import java.util.List;

/**
 *
 * @author Константин
 */
public final class DetachmentFabric {
     
       public static List<Detachment> detachments = new ArrayList<Detachment>(); 
       public static  Detachment  create(SolderInternal.SolderType type, Detachment.SizeGroupe size)
        {
           Detachment newd = new Detachment();
           newd = new Detachment();
           newd.size = size;
           newd.type = type; 
           return newd;  
        }
       public static Detachment getDetachment(Detachment.SizeGroupe size,SolderInternal.SolderType type)
       {
           for (Detachment det : detachments) {
               if(det.type == type)
                   if(det.size ==  size)
                    return det;
           }
           
           Detachment newd =  create(type, size);
           detachments.add(newd); 
           return newd;  
       } 
    }
 
